cdap-master/src/main/java/co/cask/cdap/master/startup/ConfigurationCheck.java

```diff
@@ -191,8 +191,8 @@ private boolean isValidKafkaTopic(String key) {
 
   private boolean isValidPartitionKey(String key) {
     try {
-      LogPartitionType.valueOf(key);
-    } catch (IllegalArgumentException e) {
+      LogPartitionType.valueOf(key.toUpperCase());
+    } catch (IllegalArgumentException | NullPointerException e) {
```

Comment:
I just wonder how can that happen, as it might indicate some other error or otherwise we have to check for `null` for every places we call `cConf.get` (which is pretty ugly). Did you have an up-to-date cdap-common jar, as it can only happen when the `cdap-default.xml` inside the `cdap-comomon.jar` doesn't contain the key.